client/src/components/InputsModal/InputsModal.tsx

```diff
@@ -133,22 +136,131 @@ const InputsModal: FC<InputsModalProps> = ({
     }
   });
 
+  const [inputType, setInputType] = React.useState<string>('Field');
+  const [baseInput, setBaseInput] = React.useState<string>('');
+  const [invalidInput, setInvalidInput] = React.useState<boolean>(false);
+
+  useEffect(() => {
+    const serialObj = inputs.map((requirement: TestInput) => {
+      if (requirement.type == 'oauth_credentials') {
+        return {
+          ...requirement,
+          value:
+            (JSON.parse(inputsMap.get(requirement.name) as string) as OAuthCredentials) || '{}',
+        };
+      } else if (requirement.type == 'radio') {
+        const firstVal =
+          requirement.options?.list_options && requirement.options?.list_options?.length > 0
```

Comment:
```suggestion
          requirement.options?.list_options?.length > 0
```

Since you're already employing optional chaining I think you can simply this. If the property isn't present `undefined` will be returned and `undefined > 0` will always return false.

Reply:
I actually think this causes a type error in TS since `undefined` can't evaluate to a number. Haven't checked in this particular case but we may want to keep it as is if this is true.

Reply:
Yeah, it causes a type error.